### DISCUSSION TITLE
[vite] Only using this plugin for message compilation?

### DISCUSSION THREAD

**Discussion by MartinMalinda** [intlify/bundle-tools#109] [Q&A]
Hi!

Thanks for vue-i18n in general. I'm already using it in my app and I'm happy with it. I'm actually not using this vite plugin. I integrated vue-i18n directly.

I'm wondering if I should eventually migrate to this solution. Would there be performance benefits if I'd do?

At this point I'd only benefit from extra tree shaking and also the precompilation of locales. I was looking for some easy way for the precompilcation, some kind of simple input output operation (point it to existing files, let it transform them in place etc). But I could not get that going. It feels like I have to go all in when I use this plugin.

I'm actually even not sure if this vite plugin does the precompilation. Only let it point to the locale files and let them transform in place.

EDIT: If I try to use the plugin it crashes on build for me as it tries to read and process al `.vue` files. I don't to process any `.vue` files at all. I only need to precompile the locale files 🙏 so I can bundle the production module.

Thanks!

**Comment by kazupon** [comment]
You will almost never have to configure pre-compiling of locale messages or tree shaking with vite-plugin-vue-i18n. The plugin will also do these tedious configurations for you, depending on your vite development/production environment. It then configures the runtime and runtime-less versions of vue-i18n internally.

**Reply by MartinMalinda** [reply]
Thank you for the reply. Unfortunately I can't get that the plugin to work normally since I get an error during build `[vite]: Rollup failed to resolve import "/~/utils/log" from "src/main.ts".` Seems like path aliases stop working ever since I add `vite-i18n` to plugins.

Is there some function I could use that would work in a simple input output way? I'd have no issues adding this to my build system. Basically replacing locale files at the right time is not an issue for me. But I'm afraid making the whole plugin work for me would be a lot of work. I also don't need `<i18n>` component to work etc. I also lazy load the language on startup in a very custom way and I'm not sure if I could do that with the normal workflow?
